framework/Models/PostProcessors/BasicStatistics.py

```diff
@@ -963,17 +983,16 @@ def __runLocal(self, inputData):
         for target in needed[metric]['targets']:
           targWeight = relWeight[target].values
           targDa = dataSet[target]
-          quantile = []
-          for pct in percent:
-            if self.pivotParameter in targDa.sizes.keys():
-              qtl = [self._computeWeightedPercentile(group.values,targWeight,percent=pct) for label,group in targDa.groupby(self.pivotParameter)]
-            else:
-              qtl = self._computeWeightedPercentile(targDa.values,targWeight,percent=pct)
-            quantile.append(qtl)
           if self.pivotParameter in targDa.sizes.keys():
-            da = xr.DataArray(quantile,dims=('percent',self.pivotParameter),coords={'percent':percent,self.pivotParameter:self.pivotValue})
+            quantile = []
+            for label, group in targDa.groupby(self.pivotParameter):
+              qtl = self._computeWeightedPercentile(group.values, targWeight, percent=percent)
+              quantile.append(qtl)
+            da = xr.DataArray(quantile, dims=(self.pivotParameter, 'percent'), coords={'percent': percent, self.pivotParameter: self.pivotValue})
           else:
-            da = xr.DataArray(quantile,dims=('percent'),coords={'percent':percent})
+            quantile = self._computeWeightedPercentile(targDa.values, targWeight, percent=percent)
+            da = xr.DataArray(quantile, dims=('percent'), coords={'percent': percent})
```

Comment:
Yes, the calculation of quantiles originally required an argsort every time a new percentile was calculated. I moved the argsort outside of this so that the data is sorted once and then multiple quantiles are calculated from that sorted data. This scaled roughly linearly with the number of percents requested.

Reply:
The calculation of quantiles was slowed down by using "_computeWeightedPercentile" instead of using native xarray.DataSet.quantile() methods. I changed this so that if there are no probability weights present or if all the probability weights are equal native xarray.DataSet.quantile() is used. This speeds up the calculation of percentiles significantly. At this time, xarray does not have support for weighted quantiles, so if probability weights are present and they are not all the same, we have to use "_computeWeightedPercentile." The new formulation of "_computeWeightedPercentile" is faster than before, but is still slower than native xarray.DataSet.quantile().